### PR TITLE
Improve regex printing

### LINKED ISSUE #1334: Flow parser add escapes to regexes and normalizes flags
```js
/[/]/ig;
```

outputs

```js
/[\/]/gi;
```

https://prettier.github.io/prettier/#%7B%22content%22%3A%22%2F%5B%2F%5D%2Fig%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22all%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22flow%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D

### PR DESCRIPTION
**Pull request by lydell**
- Print the raw regex for Flow, just like for Babylon.
- Sort regex flags.

Fixes #1334.

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -878,11 +878,13 @@ function genericPrintNoParens(path, options, print, args) {
     case "ThisExpression":
       return "this";
     case "Super":
-      return "super"; // Babel 6 Literal split
+      return "super";
+    // Babel 6 Literal split
```
Comment (SimenB) [inline src/printer.js]: Should this be above `"super"`? That's what I'd think looking at the old version

Comment (lydell) [inline src/printer.js]: My guess is that I have restored the original comment position after Prettier had moved it to the end of the previous line some time.

**Comment by lydell** [comment]
I don’t understand why the test fails in Node.js 4.

File: src/printer.js (hunk 1 above)
Comment (SimenB) [inline src/printer.js]: Aye, probably 😄

**Comment by lydell** [comment]
The tests should pass now. Apparently, Flow fails to parse regexes with the `u` or `y` flags in Node.js 4, so I simply removed them from the test.

```
> require("flow-parser").parse("/a/u")
0,248,Js.Error,6,SyntaxError: Invalid flags supplied to RegExp constructor 'u'
```

**Comment by vjeux** [comment]
Thanks!

**Pull request merged**